packages/validator/templates/OAS2OIS.json

```diff
@@ -0,0 +1,288 @@
+{
+	"__actions": [
+		{
+			"__insert": {
+				"__target": "oisFormat",
+				"__value": "1.0.0"
+			}
+		},
+		{
+			"__insert": {
+				"__target": "apiSpecifications.servers",
+				"__value": []
+			}
+		},
+		{
+			"__insert": {
+				"__target": "endpoints",
+				"__value": []
+			}
+		},
+		{
+			"__insert": {
+				"__target": "apiSpecifications.security",
+				"__value": {}
+			}
+		},
+		{
+			"__insert": {
+				"__target": "apiSpecifications.components.securitySchemes",
+				"__value": {}
+			}
+		},
+		{
+			"__insert": {
+				"__target": "apiSpecifications.paths",
+				"__value": {}
+			}
+		}
+	],
+	"info": {
+		"title": {
+			"__actions": [
+				{
+					"__copy": {
+						"__target": "title"
+					}
+				}
+			]
+		},
+		"version": {
+			"__actions": [
+				{
+					"__copy": {
+						"__target": "version"
+					}
+				}
+			]
+		},
+		"__ignore": {}
+	},
+	"servers": {
+		"__arrayItem": {
+			"url": {
+				"__regexp": "^(https?|ftp)://[^\\s/$.?#].[^\\s]*$",
+				"__actions": [
+					{
+						"__copy": {
+							"__target": "apiSpecifications.servers[0].url"
+						}
+					}
+				]
+			}
+		}
+	},
+	"paths": {
+		"__keyRegexp": "^\\/[^\\s'\"\\\\]+$",
```

Comment:
What the reason for the substring `\\\\` in the regex? Doesn't everything in `[^...]` get treated as a single character? i.e. `[^\\]` has the same effect as `[^\\\\]`.

(Exclaimer: I haven't written regex in Javascript, although I do imagine it's mostly the same as Python, which I'm most comfortable with.)

Reply:
Thank you for this, yes regular expressions in js work exactly as you say, however this is another case of validator's insufficient documentation. All `\` must be doubled in regular expressions, the reason for this is one `\` gets swallowed by template conversion from string to JSON object, so in this particular case `[^\\\\]` after template conversion into JSON object looks like this: `[^\\]`, which is the regular expression that we want. I added this into the [readme issue](https://github.com/api3dao/airnode/issues/203)

Reply:
but after template conversion into JSON object, wouldn't you want it to look like this: `[^\]` i.e. "not backslash", since everything in `[^...]` is treated as a single character, no?

Reply:
Yes sorry, I should have continued with the explanation. The process has one more step, after conversion from string into JSON object, it would look something like this: `{ __regexp: "[^\\]" }`, you can notice that the value is still not js regular expression, but just a normal string. So one more `\` is consumed by creation of js regular expression from string.